do not merge jsdoc tags with the same name with flag --noTagMerge

Feiyang1 opened the pull request:
Currently JSDoc tags with the same names are merged in the output externs. 
For example, 

```javascript
/**
 *  @example
 *   example 1
 *  @example
 *   example 2
*/
function test() {}
```
becomes
```javascript
/**
 * \@example
 *   example 1 /
 *   example 2
 * @return {void}
 */
function test() { }
```
Notice that @example tags are merged.

This PR adds a new flag --noTagMerge. If set to true, JSDoc tags with the same name will not be merged. This is to address an issue I opened earlier: https://github.com/angular/tsickle/issues/907
With this change, and command  `tsickle --noTagMerge --externs=output.js`
 the output will look like:
```javascript
/**
 * \@example 
 *   example 1
 * \@example 
 *   example 2
 * @return {void}
 */
function test() {}
```

I wanted to add a test case, but I couldn't find any place for module_type_translator.ts. Please let me know how I can do it properly, and I will add the test case.

mprobst commented:
Didn't look at the code yet, but:

* tests are in subdirectories of `test_files/...`, create a new folder and
a .ts file, then run `UPDATE_GOLDENS=y bazel run //test:golden_test`, then
iterate from there using `bazel test`
* make sure this doesn't break merging of e.g. an `@param` tag when there
are multiple function overloads
* I'm generally hesitant to add more flags. Could we decide on a set of
tags where we don't want the merging, and just always do the right thing?

Feiyang1 commented:
I added a test case in test/tsickle_test.ts. Is it the right place ?
I also tried to add a test file under test_files, but it always creates a js file without using the flag.
How do I tell the test suite to compile the new file with the flag I added?

mprobst commented:
See my reply by email - I'm not sure we're following the right path here.

evmar commented:
You cannot control this by a flag, because some tags *must* be merged, such as @return.  We should instead control this in tsickle -- there's no good reason for tsickle to ever merge @example tags, so we should be able to construct a whitelist of tags that should or shouldn't be merged.

evmar commented:
Should I close this?  Or do you intend to follow up on it?

Feiyang1 commented:
Please close it. I agree it is the wrong direction to go.
